### PR TITLE
show states hotkeys in chili integral menu tooltips

### PR DESCRIPTION
**Pull request by danylo-dubinin**
From https://github.com/ZeroK-RTS/Zero-K/issues/3917:

Hi, I've been augmenting my UI with hotkeys, chaning and experimenting with them a lot to develop an optimal hotkey config for my playstyle. Because of that I sometimes might forget a new hotkey that I'm trying out.
If it's a State hotkey, to recall it I open Epic Menu, click Search, type the name of the action if I remember the name, and find the hotkey in the result list.

I think command tooltips can be extended to contain more info about configured hotkeys, because we all are sometimes in that moment when you are morphing your geo in 1 match out of 30, and you cannot remember what key you have binded to Low Misc. Priority. I think it's a not neccessary moment of stress so I propose extending the tooltips so they look like in the following examples

- A subset of states has hotkeys:
states

- Only toggle has a hotkey:
toggle

- States do not exist:
(no change)
short

In short, making all the hotkeys for a State command displayed in the tooltips.

Criticism is welcome.

> Sprunk:
Sounds like it would be good to make all of the states show up if any of them has a hotkey (in this example with Low: (none)).

### CODE REVIEW

**Comment by GoogleFrog** [comment]
I am slightly against the suggestion of adding '(none)' for unset hotkeys if some hotkey is set. Most (all?) of the default hotkeys for states are just toggles, not keys that set states. I want to avoid scaring people who are looking around the UI tooltips with too much information.

For consistency with the other tooltips I propose that the state toggle hotkey be shown in parenthesis at the end of the description string. If the state toggle has hotkeys that set a particular state then make the list of hotkeys with "State Hotkeys:" at the top as shown in your screenshot. This would list all the states and the hotkeys for each, including '(none)' for states that have no hotkey.

**Comment by sprunk** [comment]
> I am slightly against the suggestion of adding '(none)' for unset hotkeys if some hotkey is set. Most (all?) of the default hotkeys for states are just toggles, not keys that set states.

I agree, what I said earlier had poor wording (didn't mean to suggest Low/Normal/High appearing if just Toggle is set, just only if one of L/N/H is).

**Comment by danylo-dubinin** [comment]
Agreeing your comments @GoogleFrog and @sprunk. Updated.

- State hotkeys exist, but only toggle is set:
image

- No toggle is set, some state hotkey is set:
image

- Toggle is set, some state hotkey is set:
image

File: LuaUI/Widgets/gui_chili_integral_menu.lua [hunk 1]
```diff
@@ -478,27 +480,65 @@ local function RemoveAction(cmd, types)
...
-		if tooltip and hotkey then
-			tooltip = tooltip .. " (\255\0\255\0" .. hotkey .. "\008)"
+		local hotkey_for_toggle = GetHotkeyText(command.action)
+		local hotkeys_for_states, states_n = GetHotkeysForStatesText(command.action)
+		if hotkey_for_toggle then
+			tooltip = tooltip .. " (" .. GetGreenStr(hotkey_for_toggle) .. ")"
+		end
+		if hotkeys_for_states and states_n > 0 then
+			tooltip = tooltip .. SEP .. "State Hotkeys:"
+			for state_name, hotkey in pairs(hotkeys_for_states) do
```
Comment (GoogleFrog) [inline LuaUI/Widgets/gui_chili_integral_menu.lua:539]: I should have requested that this pairs that exists purely to iterate over a transient table be made to not exist.

Comment (danylo-dubinin) [reply LuaUI/Widgets/gui_chili_integral_menu.lua:539]: Is it a performance consideration?

Comment (GoogleFrog) [reply LuaUI/Widgets/gui_chili_integral_menu.lua:539]: Yes.
